### DISCUSSION TITLE
Translate WPF CaptureMouse() / ReleaseMouseCapture() to Avalonia

### DISCUSSION THREAD

**Discussion by llfab** [AvaloniaUI/Avalonia#7188] [Q&A] [answered]
In order to receive the `MouseMove` event in WPF you need to call `FrameworkElement.CaptureMouse()` on mouse down and `FrameworkElement.ReleaseMouseCapture()` on mouse up. 

There is no such thing in Avalonia. Does that mean, that this is somehow automatically done in Avalonia?

**Comment by maxkatz6** [comment, marked as answer]
You can capture pointer on pointer pressed event (or maybe any other pointer event as well). See args.Pointer.Capture()

**Reply by llfab** [reply]
And the release would then be `e.Pointer.Capture(null)` right?

**Reply by llfab** [reply]
I checked Avalonia source to see that apparently `Capture(null)` ist the "release" method

**Comment by kekekeks** [comment]
You don't need to capture manually, implicit pointer capture happens automatically before PointerPressed event is raised.

**Reply by llfab** [reply]
Thanks for this. So, to understand this fully: 
- to receive mouse move after mouse down (for drag&drop) I do not need to capture and "un"-capture manually
- manual capture/uncapture would only be necessary for non-default behavior
Correct?

**Reply by kekekeks** [reply]
1) it should happen automatically, yes
2) manual `Capture` calls are mostly required if you want to transfer existing implicit pointer capture to another control (for example that happens if some parent control's gesture recognizer detects a gesture and wants to take over the pointer.

**Comment by 29654761** [comment]
What shall I do if I don't want to capture the pointer when pressed?
